### PR TITLE
PMM-11334 Added migration script for dbaas

### PR DESCRIPTION
**Pull request by gen1us2k**
PMM-0

Build: SUBMODULES-0

- [ ] Links to other linked pull requests (optional).

### CODE REVIEW

**Review by recharte — APPROVED** [review]
Looks good :+1:

File: migrate-dbaas.py [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+    volume_spec = cluster.get("spec", {}).get("pxc", {}).get("volumeSpec", {}).get("persistentVolumeClaim", {})
+    limits = cluster.get("spec", {}).get("pxc", {}).get("resources", {}).get("limits", {})
+    database_cluster["spec"]["dbInstance"] = {
+        "cpu": limits.get("cpu"),
+        "memory": limits.get("memory"),
+        "diskSize": volume_spec.get("resources", {}).get("requests", {}).get("storage"),
+        "storageClassName": volume_spec.get("storageClassName", None)
+    }
+
+
```
Comment (PavelKhripkov) [inline migrate-dbaas.py:99]:
```suggestion

```

File: migrate-dbaas.py [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+    volume_spec = replsets[0].get("volumeSpec", {}).get("persistentVolumeClaim", {})
+    limits = replsets[0].get("resources", {}).get("limits", {})
+    database_cluster["spec"]["dbInstance"] = {
+        "cpu": limits.get("cpu"),
+        "memory": limits.get("memory"),
+        "diskSize": volume_spec.get("resources", {}).get("requests", {}).get("storage"),
+        "storageClassName": volume_spec.get("storageClassName", None)
+    }
+
+
```
Comment (PavelKhripkov) [inline migrate-dbaas.py:168]:
```suggestion

```

File: migrate-dbaas.py [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+def get_clusters(cluster_type):
+    output = subprocess.check_output(["kubectl", "get", cluster_type, "-o", "json"])
+    return json.loads(output)
+
+
+def create_cluster(cluster):
+    p = subprocess.Popen(["kubectl", "apply", "-f", "-"], stdin=subprocess.PIPE)
+    out = p.communicate(json.dumps(cluster).encode('utf-8'))
```
Comment (PavelKhripkov) [inline migrate-dbaas.py:13]:
```suggestion
    p.communicate(json.dumps(cluster).encode('utf-8'))
```
Unused.

**Review by PavelKhripkov — APPROVED** [review]

**Pull request merged**